app/src/pages/inside/dashboardItemPage/newWidgets/c3chart/c3chart.jsx

```diff
@@ -0,0 +1,79 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import c3 from 'c3';
+
+export class C3Chart extends React.Component {
+  static propTypes = {
+    config: PropTypes.object,
+    onChartCreated: PropTypes.func,
+    className: PropTypes.string,
+    style: PropTypes.object,
+  };
+
+  static defaultProps = {
+    config: {},
+    onChartCreated: () => {},
+    className: '',
+    style: {},
+  };
+
+  componentDidMount() {
+    this.updateChart(this.props.config);
+  }
+
+  componentWillReceiveProps(newProps) {
+    this.updateChart(newProps.config);
+  }
+
+  componentWillUnmount() {
+    this.destroyChart();
+  }
+
+  destroyChart() {
+    try {
+      this.chart = this.chart.destroy();
+    } catch (err) {
+      throw new Error('Internal C3 error', err);
+    }
```

Comment:
Maybe will be better to use something like this.chart && this.chart.destroy()?

Reply:
Done